### PR TITLE
bpo-35384: The repr of ctypes.CArgObject no longer fail for non-ascii character.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue35384

### CODE REVIEW

File: Modules/_ctypes/callproc.c [hunk 1]
```diff
@@ -514,8 +522,8 @@ PyCArg_repr(PyCArgObject *self)
         break;
 
     default:
-        sprintf(buffer, "<cparam '%c' at %p>",
-            self->tag, self);
+        sprintf(buffer, "<cparam %02x' at %p>",
```
Comment (gareth-rees) [inline Modules/_ctypes/callproc.c]: Looks as if this change accidentally removed the opening apostrophe.
```suggestion
        sprintf(buffer, "<cparam '%02x' at %p>",
```

File: Modules/_ctypes/callproc.c [hunk 2]
```diff
@@ -498,9 +498,17 @@ PyCArg_repr(PyCArgObject *self)
         break;
 
     case 'c':
-        sprintf(buffer, "<cparam '%c' (%c)>",
-            self->tag, self->value.c);
-        break;
+        if ((unsigned char)self->value.c < 128
+            && _PyUnicode_IsPrintable(self->value.c)
```
Comment (gareth-rees) [inline Modules/_ctypes/callproc.c]: Is printability the right test here? Tab, newline, carriage return, vertical tab, form feed are all printable but probably don't make sense in this context.

File: Modules/_ctypes/callproc.c [hunk 3]
```diff
@@ -498,9 +498,17 @@ PyCArg_repr(PyCArgObject *self)
...
+            && self->value.c != '\'')
```
Comment (gareth-rees) [inline Modules/_ctypes/callproc.c]: Is the goal here to make the output look like a syntactically valid Python string? If so, you should also exclude backslash.

File: Modules/_ctypes/callproc.c (hunk 2 above)
Comment (serhiy-storchaka) [reply Modules/_ctypes/callproc.c]: `'\t'.isprintable()` returns False.

File: Modules/_ctypes/callproc.c (hunk 3 above)
Comment (serhiy-storchaka) [reply Modules/_ctypes/callproc.c]: Good catch. Yes, the goal here to make the output look like a syntactically valid Python or C.

File: Modules/_ctypes/callproc.c (hunk 1 above)
Comment (serhiy-storchaka) [reply Modules/_ctypes/callproc.c]: Actually the opposite. I have forgot to remove the closing apostrophe.

**Pull request merged**